### PR TITLE
Add `CurrentRouteInterface` containing only getters for use in userland

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

Adopt PRs:
- https://github.com/yiisoft/router-fastroute/pull/78
- https://github.com/yiisoft/app/pull/180
- https://github.com/yiisoft/demo/pull/333

### CODE REVIEW

**Comment by yiiliveext** [comment]
I prefer this way https://github.com/yiisoft/router/pull/118

**Comment by vjik** [comment]
> I prefer this way #118

This PR not does not contradict to #118

**Comment by yiiliveext** [comment]
> > I prefer this way #118
> 
> This PR not does not contradict to #118

It is not necessary if we accept #118

File: src/CurrentRouteInterface.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Router;
+
+use Psr\Http\Message\UriInterface;
+
+interface CurrentRouteInterface
```
Comment (xepozz) [inline src/CurrentRouteInterface.php:9]: It's not a `route` at all. Can you please rename it to `provider`?

Comment (vjik) [reply src/CurrentRouteInterface.php:9]: Does this mean that `CurrentRoute` also should be renamed to `CurrentRouteProvider` ?

Comment (xepozz) [reply src/CurrentRouteInterface.php:9]: It would be great

**Review by samdark — APPROVED** [review]

**Pull request merged**